src/ChannelServer/World/Inventory.cs

```diff
@@ -232,12 +271,30 @@ public int AddSilent(Item item)
 		/// <return>Index of the item.</return>
 		public int Add(Item item, InventoryAddType addType)
 		{
-			var index = this.AddSilent(item);
+			Dictionary<int, int> indexes = this.AddSilent(item);
+
+			var lastItem = indexes.Last();
```

Comment:
Another tip: you can save one line here. `var lastIndex = indexes.Values.Last();`